monai/networks/nets/flexible_unet.py

```diff
@@ -9,49 +9,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import List, Optional, Sequence, Tuple, Union
+from typing import List, Optional, Sequence, Tuple, Type, Union
 
 import torch
 from torch import nn
 
-from monai.networks.blocks import UpSample
+from monai.networks.blocks import BasicEncoder, UpSample
 from monai.networks.layers.factories import Conv
 from monai.networks.layers.utils import get_act_layer
-from monai.networks.nets import EfficientNetBNFeatures
+from monai.networks.nets import EfficientNetEncoder
 from monai.networks.nets.basic_unet import UpCat
 from monai.utils import InterpolateMode
 
-__all__ = ["FlexibleUNet"]
+__all__ = ["FlexibleUNet", "BACKBONE"]
 
-encoder_feature_channel = {
-    "efficientnet-b0": (16, 24, 40, 112, 320),
-    "efficientnet-b1": (16, 24, 40, 112, 320),
-    "efficientnet-b2": (16, 24, 48, 120, 352),
-    "efficientnet-b3": (24, 32, 48, 136, 384),
-    "efficientnet-b4": (24, 32, 56, 160, 448),
-    "efficientnet-b5": (24, 40, 64, 176, 512),
-    "efficientnet-b6": (32, 40, 72, 200, 576),
-    "efficientnet-b7": (32, 48, 80, 224, 640),
-    "efficientnet-b8": (32, 56, 88, 248, 704),
-    "efficientnet-l2": (72, 104, 176, 480, 1376),
-}
 
-
-def _get_encoder_channels_by_backbone(backbone: str, in_channels: int = 3) -> tuple:
+class Register:
+    """
+    A register to regist backbones for the flexible unet. All backbones can be found in
+    register_dict.
     """
-    Get the encoder output channels by given backbone name.
 
-    Args:
-        backbone: name of backbone to generate features, can be from [efficientnet-b0, ..., efficientnet-b7].
-        in_channels: channel of input tensor, default to 3.
+    def __init__(self):
+        self.register_dict = {}
 
-    Returns:
-        A tuple of output feature map channels' length .
-    """
-    encoder_channel_tuple = encoder_feature_channel[backbone]
-    encoder_channel_list = [in_channels] + list(encoder_channel_tuple)
-    encoder_channel = tuple(encoder_channel_list)
-    return encoder_channel
+    def regist_class(self, name: Type):
```

Comment:
Please consider to support `str` as the arg, in case we use it in the JSON config of bundle.

Thanks.

Reply:
Hi @Nic-Ma ,
     I have added string input to regist_class function.
Thanks,
Bin